UI/window-basic-settings-stream.cpp

```diff
@@ -262,10 +262,11 @@ void OBSBasicSettings::SaveStream1Settings()
 		obs_data_set_bool(settings, "bwtest", false);
 	}
 
-	if (whip)
+	if (whip) {
+		obs_data_set_string(settings, "service", "WHIP");
 		obs_data_set_string(settings, "bearer_token",
 				    QT_TO_UTF8(ui->key->text()));
-	else
+	} else
```

Comment:
Would be good to have the bracket for the else statement too for code consistency.